src/middlewared/middlewared/plugins/security/info.py

```diff
@@ -0,0 +1,31 @@
+from subprocess import run
+
+from middlewared.schema import accepts, returns, Bool
+from middlewared.service import CallError, Service
+
+
+class SystemSecurityInfoService(Service):
+
+    class Config:
+        namespace = 'system.security.info'
+        cli_namespace = 'system.security.info'
+
+    @accepts(roles=['READONLY'])
+    @returns(Bool('fips_available'))
+    def fips_available(self):
+        """Returns a boolean identifying whether or not FIPS
+        mode may be toggled on this system"""
+        # being able to toggle fips mode is hinged on whether
+        # or not this is an iX licensed piece of hardware
+        return bool(self.middleware.call_sync('system.license'))
```

Comment:
Also this method can be asynchronous ?

Reply:
No real benefit to making it asynchronous

Reply:
This will result in it being run in a thread when we don't really need it to be in a thread

Reply:
it's calling a synchronous routine which runs in our thread pool no matter how we slice it.

Reply:
Ah, makes sense